### PR TITLE
Make `pre-commit` happy

### PR DESCRIPTION
**Pull request by kenodegard**
### Description

While we added `.pre-commit-config.yaml` back in Oct 2023 (https://github.com/conda-incubator/conda-dot-org/issues/167) we didn't get around to enabling pre-commit CI until just now in March 2024. Naturally fixes were necessary to make pre-commit happy again (not to mention version updates).

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -311,9 +311,11 @@ As an alternative we offer the ability to generate randomized banners overlaid w
 $ npm run banner -- --output static/img/blog/<post>/banner.png
 ```
 
+<!-- prettier-ignore-start -->
 > [!NOTE]
 > `canvas` isn't compiled for `macOS-arm64`; navigate to the following page for compiling instructions:
 > https://www.npmjs.com/package/canvas#compiling
+<!-- prettier-ignore-end -->
```
Comment (kenodegard) [inline CONTRIBUTING.md:318]: Unfortunately `prettier` doesn't understand GitHub admonitions yet (https://github.com/prettier/prettier/issues/15479)

**Comment by jakirkham** [comment]
Just merged PR ( https://github.com/conda-incubator/conda-dot-org/pull/188 ), which had some fixes and introduced some conflict here. Tried to resolve the conflicts, but please take another look to make sure this has the expected changes

**Comment by kenodegard** [comment]
@jakirkham thanks for the heads up!

**Review by jakirkham — APPROVED** [review]

**Comment by jakirkham** [comment]
Thanks Ken! 🙏

**Pull request merged**